### PR TITLE
[REVIEW] Fix issue with not setting GPU on background thread [skip ci]

### PR DESCRIPTION
**Pull request by revans2**
If a non-default GPU is used the memory cleaner thread was still on the default GPU so they were not being collected, even though all of the code indicated that it was a success.

### CODE REVIEW

File: java/src/main/java/ai/rapids/cudf/Rmm.java [hunk 1]
```diff
@@ -36,7 +36,24 @@
    * @param poolSize       The initial pool size in bytes
    * @throws IllegalStateException if RMM has already been initialized
    */
-  public static void initialize(int allocationMode, boolean enableLogging, long poolSize)
+  public static void initialize(int allocationMode, boolean enableLogging, long poolSize) {
+    initialize(allocationMode, enableLogging, poolSize, -1);
```
Comment (jlowe) [inline java/src/main/java/ai/rapids/cudf/Rmm.java]: Should this detect the device via `cudaGetDevice`?

File: java/src/main/java/ai/rapids/cudf/MemoryCleaner.java [hunk 2]
```diff
@@ -144,11 +143,35 @@ public void clean() {
...
   }
 
+  /**
+   * The default GPU as set by user threads.
+   */
+  private static volatile int defaultGpu = -1;
+  /**
+   * The default GPU as set by the cleaner thread.
+   */
+  private static volatile int currentGpuId = -1;
```
Comment (jlowe) [inline java/src/main/java/ai/rapids/cudf/MemoryCleaner.java]: I'm not sure why this is static and volatile, as it appears to only be accessed within the cleaner thread.   Can this just be  a local variable within the thread's `run()` method?

**Comment by revans2** [comment]
@jlowe great comments I think I have fixed them

**Comment by revans2** [comment]
@kkraus14 could we get this merged in?

**Pull request merged**